pkg/clusteragent/orchestrator/transformer.go

```diff
@@ -194,3 +197,173 @@ func extractDeploymentConditionMessage(conditions []v1.DeploymentCondition) stri
+		for _, address := range n.Status.Addresses {
+			msg.Status.NodeAddresses[string(address.Type)] = address.Address
+		}
+	}
+
+	// extract conditions
+	if len(n.Status.Conditions) > 0 {
+		for _, condition := range n.Status.Conditions {
+			c := &model.NodeCondition{
+				Type:    string(condition.Type),
+				Status:  string(condition.Status),
+				Reason:  condition.Reason,
+				Message: condition.Message,
+			}
+			if !condition.LastTransitionTime.IsZero() {
+				c.LastTransitionTime = condition.LastTransitionTime.Unix()
+			}
+			msg.Status.Conditions = append(msg.Status.Conditions, c)
+		}
+	}
+
+	// extract status message
+	msg.Status.Status = computeNodeStatus(n)
+
+	// extract role
+	roles := findNodeRoles(n)
+	if len(roles) > 0 {
+		msg.Roles = roles
+	}
+
+	for _, image := range n.Status.Images {
+		msg.Status.Images = append(msg.Status.Images, &model.ContainerImage{
+			Names:     image.Names,
+			SizeBytes: image.SizeBytes,
+		})
+	}
+
+	return msg
+}
+
+func extractCapacitiesAndAllocatables(n *corev1.Node, mn *model.Node) {
+	// Milli Value ceil(q * 1000), which fits to be the lowest value. CPU -> Millicore and Memory -> byte
+	supportedResourcesMilli := []corev1.ResourceName{corev1.ResourceCPU, corev1.ResourceMemory}
+	supportedResources := []corev1.ResourceName{corev1.ResourcePods}
+	setSupportedResources(n, mn, supportedResources, false)
+	setSupportedResources(n, mn, supportedResourcesMilli, true)
+}
+
+func setSupportedResources(n *corev1.Node, mn *model.Node, supportedResources []corev1.ResourceName, isMilli bool) {
+	for _, resource := range supportedResources {
+		capacity, hasCapacity := n.Status.Capacity[resource]
+		if hasCapacity && !capacity.IsZero() {
+			if isMilli {
+				mn.Status.Capacity[resource.String()] = capacity.MilliValue()
+			} else {
+				mn.Status.Capacity[resource.String()] = capacity.Value()
+			}
+		}
+		allocatable, hasAllocatable := n.Status.Allocatable[resource]
+
+		if hasAllocatable && !allocatable.IsZero() {
+			if isMilli {
+				mn.Status.Allocatable[resource.String()] = allocatable.MilliValue()
+			} else {
+				mn.Status.Allocatable[resource.String()] = allocatable.Value()
+			}
+		}
+	}
+}
+
+func extractTaints(taints []corev1.Taint) []*model.Taint {
+	modelTaints := make([]*model.Taint, 0, len(taints))
+
+	for _, taint := range taints {
+		modelTaint := &model.Taint{
+			Key:    taint.Key,
+			Value:  taint.Value,
+			Effect: string(taint.Effect),
+		}
+		if taint.TimeAdded != nil && !taint.TimeAdded.IsZero() {
+			modelTaint.TimeAdded = taint.TimeAdded.Unix()
+		}
+		modelTaints = append(modelTaints, modelTaint)
+	}
+	return modelTaints
+}
+
+// computeNodeStatus is mostly copied from kubernetes to match what users see in kubectl
+// in case of issues, check for changes upstream: https://github.com/kubernetes/kubernetes/blob/1e12d92a5179dbfeb455c79dbf9120c8536e5f9c/pkg/printers/internalversion/printers.go#L1410
+func computeNodeStatus(n *corev1.Node) string {
```

Comment:
this would deserve unit test

Reply:
Makes sense 👍  Will also add one for `findNodeRoles`